src/mpc-hc/PPageSubMisc.cpp

```diff
@@ -92,50 +157,113 @@ BOOL CPPageSubMisc::OnApply()
 
 BEGIN_MESSAGE_MAP(CPPageSubMisc, CPPageBase)
     ON_BN_CLICKED(IDC_BUTTON1, OnBnClickedResetSubsPath)
-    ON_BN_CLICKED(IDC_BUTTON2, OnBnClickedTestSubsDB)
-    ON_UPDATE_COMMAND_UI(IDC_BUTTON2, OnUpdateButtonTestSubsDB)
-    ON_CBN_EDITCHANGE(IDC_COMBO1, OnURLModified)
+    ON_BN_CLICKED(IDC_CHECK4, OnAutoDownloadSubtitlesClicked)
+    ON_NOTIFY(NM_RCLICK, IDC_LIST1, OnRightClick)
+    ON_NOTIFY(LVN_ITEMCHANGED, IDC_LIST1, OnItemChanged)
 END_MESSAGE_MAP()
 
-void CPPageSubMisc::OnBnClickedResetSubsPath()
+void CPPageSubMisc::OnRightClick(NMHDR* pNMHDR, LRESULT* pResult)
 {
-    m_szAutoloadPaths = DEFAULT_SUBTITLE_PATHS;
+    LPNMLISTVIEW lpnmlv = (LPNMLISTVIEW)pNMHDR;
 
-    UpdateData(FALSE);
-    SetModified();
-}
+    if (lpnmlv->iItem >= 0 && lpnmlv->iSubItem >= 0) {
+        SubtitlesProvider& provider = *(SubtitlesProvider*)(m_list.GetItemData(lpnmlv->iItem));
 
-void CPPageSubMisc::OnBnClickedTestSubsDB()
-{
-    CString ISDb, ver, str;
-    UINT msg;
+        enum {
+            SET_CREDENTIALS = 0x1000,
+            RESET_CREDENTIALS,
+            MOVE_UP,
+            MOVE_DOWN,
+            OPEN_URL,

```

Comment:
I won't go into all redundant commas, just leaving a comment here so that it reminds you to get rid of them. :)